### PR TITLE
fixing new elo cards

### PR DESCRIPTION
**Pull request by lucianosousa**

### CODE REVIEW

**Comment by Fivell** [comment]
@lucianosousa ,what are you trying to do ?

**Comment by lucianosousa** [comment]
@Fivell updating the rules for elo bins
they just changed a lot of stuffs about it
also, discover card had a precedence on yml file that skips elo validations
this line basicaly:
```
- '65' 
```
Elo card now have a lot of bins started with this value
what I did with the range was also Elo `suggestion` they now have a range of bin values
I'm still working on it here today. Hope finish this issue soon.

Any suggestion about the code?

**Comment by Fivell** [comment]
@lucianosousa ,  you're trying to describe regexp in yaml file for bin, like

`   - '6500(3[5-9]|4[0-9]|5[0-1])'`

I think it can not be supported out of the box, regexp is builded by prefixes  here 486329e70e41e7ef8647daa1daf806a06c17d4d3

**Comment by Fivell** [comment]
@lucianosousa , also please take a look here https://github.com/Fivell/credit_card_validations/blob/master/lib/credit_card_validations/factory.rb#L23


This means that this gem should be able to generate valid elo number. Also think it expects no regexp in yaml file

**Comment by Fivell** [comment]
- '6500(3[5-9]|4[0-9]|5[0-1])' this one can be described as

```
6500354
6500364
6500374
6500384
6500394
```

**Comment by lucianosousa** [comment]
@Fivell take a look on my last commit. I already updated these things, just put bin code on yml again

**Comment by lucianosousa** [comment]
https://github.com/Fivell/credit_card_validations/pull/75/commits/43b2193207a72098412640bef92cc1773cb695ce

**Comment by Fivell** [comment]
looks like
650034, 650030 are DISCOVER now and all other 65003* should be detected as elo

**Comment by Fivell** [comment]
@lucianosousa , it will not work for generating elo numbers

**Comment by lucianosousa** [comment]
So, this is the file they gave us:

```
BINS RANGE ELO(V.20151028)				
(#)	PRODUCT	ACCEPTANCE   START RANGE FINAL RANGE
26	CRÉDITO	NACIONAL	431274	431274
27	CRÉDITO	NACIONAL	506717	506717
28	CRÉDITO	NACIONAL	506718	506718
29	CRÉDITO	NACIONAL	506733	506733
30	CRÉDITO	NACIONAL	506739	506739
31	CRÉDITO	NACIONAL	506741	506741
32	CRÉDITO	NACIONAL	506742	506742
33	CRÉDITO	NACIONAL	506743	506743
34	CRÉDITO	NACIONAL	506774	506774
35	CRÉDITO	NACIONAL	506776	506776
36	CRÉDITO	NACIONAL	506778	506778
37	CRÉDITO	NACIONAL	509004	509004
38	CRÉDITO	NACIONAL	509005	509005
39	CRÉDITO	NACIONAL	509006	509006
40	CRÉDITO	NACIONAL	509007	509007
41	CRÉDITO	NACIONAL	509008	509008
42	CRÉDITO	NACIONAL	509009	509009
43	CRÉDITO	NACIONAL	509013	509013
44	CRÉDITO	NACIONAL	509020	509020
45	CRÉDITO	NACIONAL	509021	509021
46	CRÉDITO	NACIONAL	509022	509022
47	CRÉDITO	NACIONAL	509023	509023
48	CRÉDITO	NACIONAL	509024	509024
49	CRÉDITO	NACIONAL	509025	509025
50	CRÉDITO	NACIONAL	509026	509026
51	CRÉDITO	NACIONAL	509027	509027
52	CRÉDITO	NACIONAL	509028	509028
53	CRÉDITO	NACIONAL	509029	509029
54	CRÉDITO	NACIONAL	509031	509031
55	CRÉDITO	NACIONAL	509033	509033
56	CRÉDITO	NACIONAL	509035	509035
57	CRÉDITO	NACIONAL	509036	509036
58	CRÉDITO	NACIONAL	509037	509037
59	CRÉDITO	NACIONAL	509038	509038
60	CRÉDITO	NACIONAL	509039	509039
61	CRÉDITO	NACIONAL	509040	509040
62	CRÉDITO	NACIONAL	509041	509041
63	CRÉDITO	NACIONAL	509042	509042
64	CRÉDITO	NACIONAL	509043	509043
65	CRÉDITO	NACIONAL	509044	509044
66	CRÉDITO	NACIONAL	509045	509045
67	CRÉDITO	NACIONAL	509046	509046
68	CRÉDITO	NACIONAL	509047	509047
69	CRÉDITO	NACIONAL	509048	509048
70	CRÉDITO	NACIONAL	509049	509049
71	CRÉDITO	NACIONAL	509050	509050
72	CRÉDITO	NACIONAL	509051	509051
73	CRÉDITO	NACIONAL	509052	509052
74	CRÉDITO	NACIONAL	509053	509053
75	CRÉDITO	NACIONAL	509064	509064
76	CRÉDITO	NACIONAL	509077	509077
77	CRÉDITO	NACIONAL	509078	509078
78	CRÉDITO	NACIONAL	509079	509079
79	CRÉDITO	NACIONAL	509080	509080
80	CRÉDITO	NACIONAL	636297	636297
94	MÚLTIPLO	NACIONAL	401178	401178
95	MÚLTIPLO	NACIONAL	401179	401179
96	MÚLTIPLO	NACIONAL	438935	438935
97	MÚLTIPLO	NACIONAL	457393	457393
98	MÚLTIPLO	NACIONAL	457631	457631
99	MÚLTIPLO	NACIONAL	457632	457632
100	MÚLTIPLO	NACIONAL	504175	504175
101	MÚLTIPLO	NACIONAL	506720	506720
104	MÚLTIPLO	NACIONAL	506721	506721
105	MÚLTIPLO	NACIONAL	506724	506724
106	MÚLTIPLO	NACIONAL	506725	506725
107	MÚLTIPLO	NACIONAL	506726	506726
108	MÚLTIPLO	NACIONAL	506727	506727
109	MÚLTIPLO	NACIONAL	506728	506728
110	MÚLTIPLO	NACIONAL	506729	506729
111	MÚLTIPLO	NACIONAL	506730	506730
112	MÚLTIPLO	NACIONAL	506731	506731
113	MÚLTIPLO	NACIONAL	506732	506732
114	MÚLTIPLO	NACIONAL	506740	506740
115	MÚLTIPLO	NACIONAL	506744	506744
116	MÚLTIPLO	NACIONAL	506745	506745
117	MÚLTIPLO	NACIONAL	506746	506746
118	MÚLTIPLO	NACIONAL	506747	506747
119	MÚLTIPLO	NACIONAL	506748	506748
120	MÚLTIPLO	NACIONAL	506753	506753
121	MÚLTIPLO	NACIONAL	506775	506775
122	MÚLTIPLO	NACIONAL	506777	506777
123	MÚLTIPLO	NACIONAL	509000	509000
124	MÚLTIPLO	NACIONAL	509001	509001
125	MÚLTIPLO	NACIONAL	509002	509002
126	MÚLTIPLO	NACIONAL	509066	509066
127	MÚLTIPLO	NACIONAL	509067	509067
128	MÚLTIPLO	NACIONAL	509068	509068
129	MÚLTIPLO	NACIONAL	509069	509069
130	MÚLTIPLO	NACIONAL	509072	509072
131	MÚLTIPLO	NACIONAL	509074	509074
132	MÚLTIPLO	NACIONAL	509076	509076
133	MÚLTIPLO	NACIONAL	509081	509081
134	MÚLTIPLO	NACIONAL	509082	509082
135	MÚLTIPLO	NACIONAL	509083	509083
136	MÚLTIPLO	NACIONAL	509085	509810
137	MÚLTIPLO	NACIONAL	636368	636368
185	CRÉDITO	NACIONAL E INTERNACIONAL	650901	650920
188	MÚLTIPLO	NACIONAL E INTERNACIONAL	650485	650538
189	MÚLTIPLO	NACIONAL E INTERNACIONAL	650541	650598
190	MÚLTIPLO	NACIONAL E INTERNACIONAL	650700	650718
191	MÚLTIPLO	NACIONAL E INTERNACIONAL	650720	650727
192	MÚLTIPLO	NACIONAL E INTERNACIONAL	651652	651679
193	MÚLTIPLO	NACIONAL E INTERNACIONAL	655000	655019
194	MÚLTIPLO	NACIONAL E INTERNACIONAL	655021	655058
```

**Comment by Fivell** [comment]
@lucianosousa , see my comment above with example https://github.com/Fivell/credit_card_validations/pull/75#issuecomment-277987347

**Comment by lucianosousa** [comment]
`65003` is not an elo bin number anymore, isn't?
that's why I removed few old elo cards

**Comment by Fivell** [comment]
@lucianosousa ,  I have no information that 65003 bin was moved to another brand

**Comment by Fivell** [comment]
@lucianosousa , however there is a discussion  https://gist.github.com/erikhenrique/5931368

**Comment by Fivell** [comment]
https://gist.github.com/erikhenrique/5931368#gistcomment-1663782

**Comment by lucianosousa** [comment]
thanks for the links. I'll take a look on it here

**Comment by lucianosousa** [comment]
@Fivell in my list, there are more than 900 bins in ranges, and I really think my list is the most recent one.
on these links you sent me, there are 200 in few ranges, 80 in one, 50 in other, 50 in another, etc.

do you think it's a good idea put all these values at `brands.yml`?
do you have a better idea about my range list approach?

**Comment by Fivell** [comment]
@lucianosousa , this is now only way to make work both validation and generation numbers. 
in other case I don't know how to fix this.

**Comment by lucianosousa** [comment]
I'll think here. Number generation/validation is already working. the problem with generate numbers is that it will not create a number with the starts in the range, I got it and i'll try improve it

**Comment by Fivell** [comment]
@lucianosousa , please think about placing all bins to ``` brands.yaml```
I do belive there should be only one place to contain all bins information

**Comment by lucianosousa** [comment]
@Fivell even if there are plus than 1000 bins?

**Comment by Fivell** [comment]
@lucianosousa , yes, I think it is bad idea to place bins in  both yaml file and ruby code.

**Comment by lucianosousa** [comment]
@Fivell done.
hope it works :)

also, I can merge all commits only in one if you want

**Comment by Fivell** [comment]
@lucianosousa , thanks for PR, good job! please squash commits to make it possible to review

**Comment by Fivell** [comment]
@lucianosousa , also I want to ask to make one more improvements before merge.

in such cases 

```
 '  -  506700'
     - '506701'
     - '506702'
     - '506703'
     - '506704'
     - '506705'
     - '506706'
     - '506707'
     - '506708'
     - '506709'
```


you can just put 1 string 

```
 '   -  50670'  
```

that will include all the range ```50670(0..9)```

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,56 @@
+PATH
```
Comment (Fivell) [inline Gemfile.lock]: please remove this file

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+coverage/
```
Comment (Fivell) [inline .gitignore]: remove this from PR please

File: lib/data/brands.yaml [hunk 3]
```diff
@@ -248,16 +235,157 @@
...
+    - '506748'
+    - '506753'
+    - '506774'
+    - '506775'
+    - '506776'
+    - '506777'
+    - '506778'
     - '506699'
-    - '5067'
+    - '506700'
```
Comment (Fivell) [inline lib/data/brands.yaml]: @lucianosousa , also I want to ask to make one more improvement might be done

in such cases

 '  -  506700'
     - '506701'
     - '506702'
     - '506703'
     - '506704'
     - '506705'
     - '506706'
     - '506707'
     - '506708'
     - '506709'
you can just put 1 string

 '   -  50670'  
that will include all the range 50670(0..9)

**Comment by lucianosousa** [comment]
done @Fivell

**Comment by Fivell** [comment]
@lucianosousa , please make optimisations to  bins that you want to add. I haven't check all the list, but each bins range that ends in 0..9 can be short-cutted, see my comments above.

**Comment by lucianosousa** [comment]
@Fivell yes, find some data that could be improved

**Comment by Fivell** [comment]
@lucianosousa , ping me when ready to merge 💳

**Comment by lucianosousa** [comment]
@Fivell sorry, it's done

**Comment by Fivell** [comment]
@lucianosousa what about '509000'-'509009'

**Comment by Fivell** [comment]
506730-506739 = 50673 etc

**Comment by lucianosousa** [comment]
@Fivell update. done

**Comment by Fivell** [comment]
@lucianosousa , thanks

**Comment by lucianosousa** [comment]
cheers, bro

**Comment by Fivell** [comment]
3.4.0 🚀

**Pull request merged**